09-features.md

```diff
@@ -19,32 +19,35 @@ for use of the channel, so the presentation of those features depends
+| 0/1   | `option_data_loss_protect`        | Requires or supports extra `channel_reestablish` fields   | IN       |                           | [BOLT #2][bolt02-retransmit]                                          |
+| 3     | `initial_routing_sync`            | Sending node needs a complete routing information dump    | I        |                           | [BOLT #7][bolt07-sync]                                                |
+| 4/5   | `option_upfront_shutdown_script`  | Commits to a shutdown scriptpubkey when opening channel   | IN       |                           | [BOLT #2][bolt02-open]                                                |
+| 6/7   | `gossip_queries`                  | More sophisticated gossip control                         | IN       |                           | [BOLT #7][bolt07-query]                                               |
+| 8/9   | `var_onion_optin`                 | Requires/supports variable-length routing onion payloads  | IN9      |                           | [Routing Onion Specification][bolt04]                                 |
+| 10/11 | `gossip_queries_ex`               | Gossip queries can include additional information         | IN       | `gossip_queries`          | [BOLT #7][bolt07-query]                                               |
+| 12/13 | `option_static_remotekey`         | Static key for remote output                              | IN       |                           | [BOLT #3](03-transactions.md)                                         |
+| 14/15 | `payment_secret`                  | Node supports `payment_secret` field                      | IN9      | `var_onion_optin`         | [Routing Onion Specification][bolt04]                                 |
+| 16/17 | `basic_mpp`                       | Node can receive basic multi-part payments                | IN9      | `payment_secret`          | [BOLT #4][bolt04-mpp]                                                 |
+| 18/19 | `option_support_large_channel`    | Can create large channels                                 | IN       |                           | [BOLT #2](02-peer-protocol.md#the-open_channel-message)               |
+| 20/21 | `option_anchor_outputs`           | Anchor outputs                                            | IN       | `option_static_remotekey` | [BOLT #3](03-transactions.md)                                         |
+| 22/23 | `option_anchors_zero_fee_htlc_tx` | Anchor commitment type with zero fee HTLC transactions    | IN       | `option_static_remotekey` | [BOLT #3][bolt03-htlc-tx], [lightning-dev][ml-sighash-single-harmful] |
+| 24/25 | `option_route_blinding`           | Node supports blinded paths                               | IN9      | `var_onion_optin`         | [BOLT #4](bolt04-route-blinding)                                      |
```

Comment:
```suggestion
| 24/25 | `option_route_blinding`           | Node supports blinded paths                               | IN       | `var_onion_optin`         | [BOLT #4](bolt04-route-blinding)                                      |
```
Why make it an invoice feature? BOLT 11 does not support blinded routes and BOLT 12 already requires it.

Reply:
We could remove it, but:

- conceptually, this is an invoice feature, as it impacts how the sender is supposed to pay the invoice
- it doesn't hurt to make it an invoice feature
- it is more future-proof if we ever end up adding blinded paths to Bolt 11 invoice (not sure that will happen, but it doesn't hurt to be ready for it)?

It's also important for future upgradability: if we have to release a route blinding v2 (because we want to make some backwards-incompatible changes), even Bolt 12 invoices will have to signal which version of route blinding they want the sender to use, and a feature bit is the standard mechanism for that. It would be weird to have a `option_route_blinding_2` be an invoice feature while `option_route_blinding` wouldn't be one.

Reply:
For now, this feature bit is necessary to advertise to other nodes that they can use you in their blinded routes. It's not used to tell the sender how to pay the invoice as this is already determined by the kind of invoice.
But your right that it would make upgrading to blinding route v2 easier.

Reply:
> We could remove it, but:
> 
> * conceptually, this is an invoice feature, as it impacts how the sender is supposed to pay the invoice
> * it doesn't hurt to make it an invoice feature
> * it is more future-proof if we ever end up adding blinded paths to Bolt 11 invoice (not sure that will happen, but it doesn't hurt to be ready for it)?
> 

Wouldn't this require adding a new tag to BOLT 11 for the blinded paths? Seems it would be better to add the feature flag to the BOLT 11 context if / when that ever happens.

> It's also important for future upgradability: if we have to release a route blinding v2 (because we want to make some backwards-incompatible changes), even Bolt 12 invoices will have to signal which version of route blinding they want the sender to use, and a feature bit is the standard mechanism for that. It would be weird to have a `option_route_blinding_2` be an invoice feature while `option_route_blinding` wouldn't be one.
> 

BOLT 12 invoice features have their own context, IIUC. (i.e., not all BOLT 11 features are meaningful for BOLT 12 nor have the same dependencies).

> Also, Bolt 12 invoices can be used outside of offers (we discussed serving them in LN-url flows).

Is this still the case as per recent spec meeting discussions? My understanding is that this format is not defined.

Reply:
> Wouldn't this require adding a new tag to BOLT 11 for the blinded paths? Seems it would be better to add the feature flag to the BOLT 11 context if / when that ever happens.

> BOLT 12 invoice features have their own context, IIUC. (i.e., not all BOLT 11 features are meaningful for BOLT 12 nor have the same dependencies).

Those are good points for removing `9` from the feature context. It would bother me a bit, because this is conceptually an invoice feature and I like making things explicit, but I get that saving a few bytes in QR codes may be worth it...let's discuss this during the spec meeting!

> Is this still the case as per recent spec meeting discussions? My understanding is that this format is not defined.

Indeed there has been some back-and-forth on this, and for now it looks like people are leaning towards removing this option from the roadmap (but who knows, maybe we'll change our mind in a few months based on user feedback?).